state_transfer/endpoint/route/route.go

```diff
@@ -162,19 +156,19 @@ func (r *RouteEndpoint) createRoute(c client.Client) error {
 			Labels:    r.Labels(),
 		},
 		Spec: routev1.RouteSpec{
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromInt(int(r.Port())),
+			},
 			To: routev1.RouteTargetReference{
 				Kind: "Service",
 				Name: r.NamespacedName().Name,
 			},
-			Port: &routev1.RoutePort{
-				TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: r.Port()},
-			},
 			TLS: termination,
 		},
 	}
 
 	err := c.Create(context.TODO(), &route, &client.CreateOptions{})
-	if err != nil {
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
```

Comment:
Do we want to error if it already exists?

Reply:
@shawn-hurley The problem is that we create svc -> route serially. If service creation fails due to AlreadyExists, it will always return with that error and never go to route creation step.

Reply:
same as above, add a todo perhaps?

Reply:
That makes sense +1